snowfall/models/transformer.py

```diff
@@ -87,6 +91,9 @@ def __init__(self, num_features: int, num_classes: int, subsampling_factor: int
         else:
             self.decoder_criterion = None
 
+        # Reference: https://github.com/espnet/espnet/blob/master/espnet2/asr/ctc.py#L37
+        self.ctc_loss_fn = torch.nn.CTCLoss(reduction='none')
```

Comment:
Let's not retain this pattern, we can make the CTC loss part of the training code.  This may not carry over so easily to k2 code, and in any case is a little inflexible for our purposes.